actions/transactions/outlines_record_test.go

```diff
@@ -57,6 +59,16 @@ func TestOutlinesRecordOpReturn(t *testing.T) {
 
 		// then:
 		then.Response(res).IsOK()
+
+		// when:
+		res, _ = client.R().
+			SetHeader("Content-Type", "application/json").
+			Get(fmt.Sprintf("%s/%s/0", getTransactionDataURL, txSpec.ID()))
```

Comment:
I would rather start with endpoint for listing data recorded by user.
Most important is this 'recorded by user' because, what you already mentioned, there is no possibility to do that because data is not pinned to user. But IMHO we want to index data for given user and only he can load this data as this is what he is interested in, while other users are interested in their own data.

Reply:
1. About the 'recorded by user' thing - I've added `xpub_id` column to `Data` table. So now, `record_outline` method and getData requires `xpub_id` of a user who actually do the action.
2. I would stick to the endpoint which gets data by outpoint (txID & vout) because it seems to me that it's the most basic operation and - at the same time - allows to test the whole flow. 
3. Of course, the listings (or search) can be also introduced but - I would not return the "data" in the list items: only txIds and vouts (and maybe some other info)